tools/python/m2k-osc/ain_tab.py

```diff
@@ -0,0 +1,199 @@
+        if ain_ch0_var.get() ==1:
+            ain.enableChannel(0,True)
+            print(ain.getRange(0))
+        else:
+            ain.enableChannel(0,False)
+    def enable_ch1(self):
+        self.get_ain()
+        if ain_ch1_var.get() ==1:
+            ain.enableChannel(1,True)
+        else:
+            ain.enableChannel(1,False)
+    def get_ain(self):
+        global ain
+        if ctx_tab.ain is not None:
+            ain=ctx_tab.ain
+            self.start_thread()
+    def plot(self):
+        global instream,samples_queuech0,samples_queuech1
+        y_ch0=[]
+        y_ch1=[]
+        if samples_queuech0:
+           y_ch0=samples_queuech0.pop()
+        if samples_queuech1:
+            y_ch1=samples_queuech1.pop()
+        x=range(len(y_ch0))
+        self.linech0.set_data(x,y_ch0)
+        self.linech1.set_data(x,y_ch1)
+        ax=self.canvas.figure.axes[0]
+        if float(aout_tab.amax_ch0.get()) >= float(aout_tab.amax_ch1.get()):
+            y_max=float(aout_tab.amax_ch0.get())+1
+        else:
+            y_max=float(aout_tab.amax_ch1.get())+1
+        if float(aout_tab.amin_ch0.get())>=float(aout_tab.amin_ch1.get()):
+            y_min=float(aout_tab.amin_ch1.get())-1
+        else:
+            y_min=float(aout_tab.amin_ch0.get())-1
+
+        ax.set_xlim(0,len(x))
+        ax.set_ylim(y_min,y_max)
+        if instream is True:
+            ax.set_xlabel("Sample Count")
+            ax.set_ylabel("Gathered Data")
+        else:
+            ax.set_xlabel("Previous Sample Count")
+            ax.set_ylabel("Previous Gathered Data")
+        self.canvas.draw()
+    def set_kernelbuff(self):
+        count=int(self.kernelbuff_entry.get())
+        ain.setKernelBuffersCount(count)
+    def set_range_ch0(self):
+        if ain.isChannelEnabled(0) is True:
+            if ain_range_ch0_var.get()==1:
+                ain.setRange(0,1)
+            else:
+                ain.setRange(0,0)
+            rangelimits_ch0.set(str(ain.getRangeLimits(ain.getRange(0))))
+    def set_range_ch1(self):
+        if ain.isChannelEnabled(1) is True:
+            if ain_range_ch1_var.get()==1:
+                ain.setRange(1,1)
+            else:
+                ain.setRange(1,0)
+            rangelimits_ch1.set(str(ain.getRangeLimits(ain.getRange(1))))
+    def start_in(self):
+        global instream
+        instream=True
+        self.update_plot()
+    def start_thread(self):
+        get_samples_thread=threading.Thread(None,get_samples,"get_samples")
+        get_samples_thread.setDaemon(True)
+        get_samples_thread.start()
+    def stop_in(self):
+        global instream
+        instream=False
+    def update_plot(self):
+        self.plot()
+        if instream is True:
+            self.after(10,self.update_plot)
+    def update_readsamples(self):
+        global read_samples
+        read_samples.set(self.ain_samples_entry.get())
+        print(read_samples.get())
+        print(self.ain_samples_entry.get())
+
+
```

Comment:
nitpick: extra line

Reply:
this is no longer an issue, nor does this code section exist in the current version